cmd/utils/balance.go

```diff
@@ -0,0 +1,96 @@
+package utils
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"github.com/dymensionxyz/roller/cmd/consts"
+	"math/big"
+	"os/exec"
+	"path/filepath"
+)
+
+type ChainQueryConfig struct {
+	Denom  string
+	RPC    string
+	Binary string
+}
+
+func QueryBalance(chainConfig ChainQueryConfig, address string) (*big.Int, error) {
+	cmd := exec.Command(chainConfig.Binary, "query", "bank", "balances", address, "--node", chainConfig.RPC, "--output", "json")
+	out, err := ExecBashCommand(cmd)
+	if err != nil {
+		return nil, err
+	}
+	return ParseBalanceFromResponse(out, chainConfig.Denom)
+}
+
+func ParseBalanceFromResponse(out bytes.Buffer, denom string) (*big.Int, error) {
+	var balanceResp BalanceResponse
+	err := json.Unmarshal(out.Bytes(), &balanceResp)
+	if err != nil {
+		return nil, err
+	}
+	for _, balance := range balanceResp.Balances {
+		if balance.Denom == denom {
+			amount := new(big.Int)
+			_, ok := amount.SetString(balance.Amount, 10)
+			if !ok {
+				return nil, errors.New("unable to convert balance amount to big.Int")
+			}
+			return amount, nil
+		}
+	}
+	return big.NewInt(0), nil
+}
+
+type Balance struct {
+	Denom  string `json:"denom"`
+	Amount string `json:"amount"`
+}
+
+type BalanceResponse struct {
+	Balances []Balance `json:"balances"`
+}
+
+func VerifySequencerBalance(rollappConfig RollappConfig, requiredBalance *big.Int, insufficientBalanceErrHandler func(string) error) error {
```

Comment:
why pass a dedicated error handler vs just returning the error from this function?